### PR TITLE
Preserve Develocity build lifecycle service behavior when Isolated Projects is not enabled

### LINKED ISSUE #29013: Gradle 8.8-rc-1 is not compatible with Develocity plugin + Test Distribution
### Current Behavior

When applying Develocity plugin via a Settings plugin, ex. 

```
buildscript {
    repositories {
        mavenCentral()
        maven {
            url = uri("https://plugins.gradle.org/m2/")
        }
    }
    dependencies {
        classpath 'com.gradle:develocity-gradle-plugin:3.17.2'
    }
}


apply plugin: MyPlugin

class MyPlugin implements Plugin<Settings> {

    @Override
    void apply(Settings settings) {
        settings.pluginManager.apply(com.gradle.develocity.agent.gradle.DevelocityPlugin)
    }
}
```

And configuring the test tasks to use `testDistribution`:

```
configure(subprojects) {
    apply plugin: 'java'

    repositories {
        mavenCentral()
    }

    dependencies {
        testImplementation 'org.junit.jupiter:junit-jupiter:5.10.2'
        testImplementation 'org.junit.platform:junit-platform-launcher:1.10.2'
    }

    tasks.withType(Test) {
        useJUnitPlatform()
        develocity {
            testDistribution {
                enabled = true
                remoteExecutionPreferred = false
                maxLocalExecutors = 1
                maxRemoteExecutors = 0
            }
        }
    }
}
```

results in:

```

* Where:
Build file '/Users/rperezalcolea/Projects/github/rpalcolea/develocity-gradle-8_8-rc-1-issue/build.gradle' line: 32

* What went wrong:
A problem occurred evaluating root project 'develocity-gradle-8_8-issue'.
> Could not find method testDistribution() for arguments [build_f1zxr60n0efii4e7esqgee7dt$_run_closure1$_closure4$_closure6$_closure7@88f523c] on extension 'develocity' of type com.gradle.develocity.agent.gradle.internal.a_Decorated.

* Try:
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org.

```

It looks like the `Test` task is being decorated with `com.gradle.develocity.agent.gradle.internal.DevelocityConfigurationInternal` instead of `com.gradle.develocity.agent.gradle.internal.test.DevelocityTestConfigurationInternal` 

### Expected Behavior

It should decorate the `Test` task using `DevelocityTestConfiguration` and `DevelocityTestConfigurationInternal` accordingly

### Context (optional)

_No response_

### Steps to Reproduce

reproducer in https://github.com/rpalcolea/gradle-enterprise-plugin-test-distribution-issue/tree/main

### Gradle version

8.8-rc.1

### Gradle version that used to work

8.7

### Build scan URL (optional)

_No response_

### Your Environment (optional)

_No response_

### PR DESCRIPTION
**Pull request by bamboo**
Fixes #29013 

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by bamboo** [comment]
@bot-gradle merge

**Comment by bamboo** [comment]
@bot-gradle merge

File: platforms/enterprise/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/DevelocityBuildLifecycleServiceIntegrationTest.groovy [hunk 1]
```diff
@@ -50,6 +52,41 @@ class DevelocityBuildLifecycleServiceIntegrationTest extends AbstractIntegration
         outputDoesNotContain("Configuring ':included")
     }
 
+    @IgnoreIf(
```
Comment (wolfs) [inline platforms/enterprise/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/DevelocityBuildLifecycleServiceIntegrationTest.groovy:55]: We should use `Requires` with a precondition class, so we check that this test actually runs somewhere. In the mid-term we want to get rid of `@IgnoreIf`.

File: platforms/enterprise/enterprise/src/main/java/org/gradle/internal/enterprise/impl/DefaultDevelocityBuildLifecycleService.java [hunk 2]
```diff
@@ -19,19 +19,39 @@
...
+        // - `allprojects` executes eagerly before any project has been evaluated, allowing its effects
+        //   to be observable from other eager configuration blocks (e.g., `subprojects { ... }`).
+        // - `lifecycle.beforeProject` executes just before each project is evaluated. Therefore, its effects
+        //   are not observable from eager configuration blocks, which would anyway be incompatible with
+        //   Isolated Projects.
+        if (isIsolatedProjects()) {
+            gradle.getLifecycle().beforeProject(action::execute);
+        } else {
+            gradle.allprojects(action);
+        }
```
Comment (wolfs) [inline platforms/enterprise/enterprise/src/main/java/org/gradle/internal/enterprise/impl/DefaultDevelocityBuildLifecycleService.java:51]: 🤔 That is not great, given that we exercise the new code path only when isolated projects is on. I suppose there wasn't a way to make this work for both cases easily. Do you have a plan to do some changes in 8.9 so we also go through `lifecycle.beforeProject()`?

Comment (bamboo) [reply platforms/enterprise/enterprise/src/main/java/org/gradle/internal/enterprise/impl/DefaultDevelocityBuildLifecycleService.java:51]: We are currently working on running `isolatedProjectsIntegTest` on CI which will exercise this code path.

File: platforms/enterprise/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/DevelocityBuildLifecycleServiceIntegrationTest.groovy (hunk 1 above)
Comment (bamboo) [reply platforms/enterprise/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/DevelocityBuildLifecycleServiceIntegrationTest.groovy:55]: Done in 0cb380f042a2b4f746392b0880a789264365664b

File: platforms/enterprise/enterprise/src/main/java/org/gradle/internal/enterprise/impl/DefaultDevelocityBuildLifecycleService.java (hunk 2 above)
Comment (bamboo) [reply platforms/enterprise/enterprise/src/main/java/org/gradle/internal/enterprise/impl/DefaultDevelocityBuildLifecycleService.java:51]: See 050e5b41489b27e072873946d2cd34918bdf345c

**Pull request merged**